File: PWR055/example-forall.f90
! PWR055: consider applying offloading parallelism to forall loop
! https://www.codee.com/catalog/pwr055/
! Forall pattern
! https://www.codee.com/catalog/patterns/forall/

SUBROUTINE example(D, X, Y, n, a)
    integer, intent(in) :: n
    double precision, intent(in) :: a
    double precision, dimension (1:n), intent(in) :: X, Y
    double precision, dimension (1:n), intent(out) :: D
    integer :: i

    DO i=1,n
       D(i) = a * X(i) + Y(i)
    END DO 
END

File: PWR055/example-forall.cpp
// PWR055: consider applying offloading parallelism to forall loop
// https://www.codee.com/catalog/pwr055/
// Forall pattern
// https://www.codee.com/catalog/patterns/forall/

void example(double *D, double *X, double *Y, int n, double a) {
  for (int i = 0; i < n; ++i) {
    D[i] = a * X[i] + Y[i];
  }
}
